packages/ui/src/views/repo/repo-commits/repo-commits-view.tsx

```diff
@@ -29,55 +30,97 @@ interface RepoCommitsViewProps {
+  const SORT_DIRECTIONS = getSortDirections(t)
+  const filterHandlers = useFilters()
+
+  const isDirtyList = page !== 1
+
+  const handleResetFiltersAndPages = () => {
+    setPage(1)
+  }
 
   return (
     <SandboxLayout.Main>
-      <SandboxLayout.Content>
-        <Spacer size={10} />
+      <SandboxLayout.Content className="pt-7">
         <Text size={5} weight={'medium'}>
           Commits
         </Text>
         <Spacer size={6} />
         <div className="flex justify-between gap-5">
           <BranchSelector
-            onSelectBranch={props.selectBranchOrTag}
-            useRepoBranchesStore={props.useRepoBranchesStore}
-            useTranslationStore={props.useTranslationStore}
-            searchQuery={props.searchQuery}
-            setSearchQuery={props.setSearchQuery}
+            onSelectBranch={selectBranchOrTag}
+            useRepoBranchesStore={useRepoBranchesStore}
+            useTranslationStore={useTranslationStore}
+            searchQuery={searchQuery}
+            setSearchQuery={setSearchQuery}
           />
+          <Filters filterOptions={FILTER_OPTIONS} sortOptions={SORT_OPTIONS} filterHandlers={filterHandlers} t={t} />
         </div>
-        <Spacer size={5} />
 
-        {props.isFetchingCommits && <SkeletonList />}
+        <FiltersBar
+          filterOptions={FILTER_OPTIONS}
+          sortOptions={SORT_OPTIONS}
+          sortDirections={SORT_DIRECTIONS}
+          filterHandlers={filterHandlers}
+          t={t}
+        />
 
-        {!props.commitsList?.length && (
-          <NoData iconName="no-data-folder" title="No commits yet" description={['There are no commits yet.']} />
-        )}
+        <Spacer size={5} />
 
-        {!props.isFetchingCommits && props.commitsList?.length && (
-          <CommitsList
-            data={props.commitsList.map((item: TypesCommit) => ({
-              sha: item.sha,
-              parent_shas: item.parent_shas,
-              title: item.title,
-              message: item.message,
-              author: item.author,
-              committer: item.committer
-            }))}
-          />
+        {isFetchingCommits ? (
+          <SkeletonList />
+        ) : (
+          <>
+            {!commitsList?.length ? (
+              <NoData
+                withBorder
+                textWrapperClassName="max-w-[350px]"
+                iconName={isDirtyList ? 'no-search-magnifying-glass' : 'no-data-commits'}
+                title={isDirtyList ? 'No commits history' : 'No commits yet'}
+                description={[
+                  isDirtyList
+                    ? 'There isn’t any commit history to show here for the selected user, time range, or current page.'
+                    : "Your commits will appear here once they're made. Start committing to see your changes reflected."
+                ]}
+                primaryButton={
+                  isDirtyList
+                    ? { label: 'Clear filters', onClick: handleResetFiltersAndPages }
```

Comment:
Looks like the handler only sets to the first page, I take it that's a temporary measure?

Reply:
Added filter reset to `handleResetFiltersAndPages`